src/entry.ts

```diff
@@ -182,6 +200,45 @@ class Entry {
     return entry;
   }
 
+  /**
+   * Formats raw incoming request objects into a GCP structured HTTPRequest.
+   * Formats trace & span if users provided X-Cloud-Trace-Context in header.
+   * See more: https://cloud.google.com/trace/docs/setup#force-trace
+   *    "X-Cloud-Trace-Context: TRACE_ID/SPAN_ID;o=TRACE_TRUE"
+   * for example:
+   *    "X-Cloud-Trace-Context: 105445aa7843bc8bf206b120001000/1;o=1"
+   * Note: logs from middleware are already formatted.
+   *
+   * @private
+   */
+  private formatHttpRequest() {
+    const rawReq = this.metadata.httpRequest;
+    if (rawReq) {
+      // Handle raw http requests.
+      if (
+        'statusCode' in rawReq ||
+        'headers' in rawReq ||
+        'method' in rawReq ||
+        'url' in rawReq
+      )
+        this.metadata.httpRequest = makeHttpRequestData(rawReq);
+      // Infer trace & span if not user specified already.
+      if ('headers' in rawReq && rawReq.headers['x-cloud-trace-context']) {
+        const regex = /([a-f\d]+)?(\/?([a-f\d]+))?(;?o=(\d))?/;
+        const match = rawReq.headers['x-cloud-trace-context']
+          .toString()
+          .match(regex);
+        if (match) {
+          if (!this.metadata.trace && match[1]) this.metadata.trace = match[1];
+          if (!this.metadata.spanId && match[3])
+            this.metadata.spanId = match[3];
+          if (this.metadata.traceSampled === undefined && match[5])
+            this.metadata.traceSampled = match[5] === '1';
```

Comment:
I don't think there is much sense in having traceSampled value as true or false in this.metadata.traceSampled but having trace and spanId not set.
Probably we can reduce the "if" to just 
```
if (match[5])
```

Reply:
the users can still submit a header of ';o=0'. Which is a redundant but valid way to indicate both i) there's no trace/span and ii) don't sample this trace and span